callisto/delivery/matching_validators.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from collections import OrderedDict
+
+from six.moves.urllib.parse import parse_qs, urlsplit
+from django.core.exceptions import ValidationError
+from django.forms import URLField
+
+logger = logging.getLogger(__name__)
+
+
+def _clean_url(url):
+    url_field = URLField()
+    return url_field.clean(url.strip())
+
+
+def _get_url_parts(url):
+    url = _clean_url(url)
+    return urlsplit(url)
+
+
+def _get_initial_path(url_parts):
+    return url_parts[2].strip('/').split('/')[0].lower()
+
+
+generic_twitter_urls = [
+        'i',
+        'following',
+        'followers',
+        'who_to_follow',
+        'settings',
+        'search',
+        'tos',
+        'privacy',
+        'about',
+    ]
+
+
+def twitter_validation_function(value):
```

Comment:
Yea that problem doesn't block this PR at all. I'll add my thoughts to #181 in a bit